tests/equilibria/test_constraints.py

```diff
@@ -85,9 +103,9 @@ def test_constraint_weights(self):
             # Test that weights have been applied
 
             problem = TikhonovCurrentCOP(
```

Comment:
Could use a gradient based algorithm here to check gradient shapes - then new constraints could be added to this test much more easily when necessary.

Reply:
> Could use a gradient based algorithm here to check gradient shapes - then new constraints could be added to this test much more easily when necessary.

The `ThikhonovCurrentCOP` problem is already using `SLSQP` (line 779 if `opt_problem.py` defining the default optimiser used here)  that is a gradient based, so the gradient based is already covered ?